### ISSUE TITLE
tmpstr not defined in ice_init.F90

### ISSUE THREAD

**Issue by dabail10** [CICE-Consortium/CICE#798]
Looks like this code is scheduled for deprecation, but it does not compile because tmpstr is not defined.

! each task gets unique ice log filename when if test is true, for debugging
         if (1 == 0) then
            call get_fileUnit(nu_diag)
            write(tmpstr,'(a,i4.4)') "ice.log.task_",my_task
            open(nu_diag,file=tmpstr)
         endif

**Comment by eclare108213** [comment]
Just need to change tmpstr to tmpstr2.  Amazing this hasn't been caught before now.  Which compiler?

**Comment by dabail10** [comment]
This was on cheyenne intel. I can't recall if I had optimization turned off.

**Comment by dabail10** [comment]
This was within CESM, so CESMCOUPLED is turned on.

**Comment by dabail10** [comment]
I have a PR coming with the saltflux stuff and I will fix it there.

**Comment by apcraig** [comment]
I knew it was in CESM, but was just pointing out that the standalone model would not have caught this.  I just checked RASM.  The code is there and CESMCOUPLED is on.  We have a slightly older version.  tmpstr is defined there.  My guess is that when I was cleaning up compiler warning messages about unused variables and such, I removed tmpstr from the module.  This would not have been needed for standalone.  I suspect it's my fault and probably happened fairly recently.  Sorry about that.  Thanks @dabail10 for fixing this.
